src/distributor.ts

```diff
@@ -89,4 +93,35 @@ const addSigListeners = (): void => {
   Logger.mainLogger.debug('Registered signal listeners.')
 }
 
-initDistributor()
+const initDistributorMQMode = async (): Promise<void> => {
+  const rmqDataPublisher = new RMQDataPublisher()
+  const rmqHealthCheck = new RMQModeHeathCheck(rmqDataPublisher)
+  initLogger()
+  await dbstore.initializeDB(config)
+  await CheckpointDao.init()
+  await initRMQModeHttpServer(rmqHealthCheck)
+
+  rmqDataPublisher.start()
+
+  process.on('SIGTERM', async () => {
+    console.log('Exiting initDistributorMQMode on SIGTERM')
+    await rmqDataPublisher.cleanUp()
+    CheckpointDao.close()
+    process.exit(0)
+  })
+
+  process.on('SIGINT', async () => {
+    console.log('Exiting initDistributorMQMode on SIGTERM')
+    await rmqDataPublisher.cleanUp()
+    CheckpointDao.close()
```

Comment:
Same here, I think we may also add closeDatabase function for distributor db
(wait dbstore.closeDatabase())

Reply:
Same as above